sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/MapElements.java

```diff
@@ -85,42 +85,57 @@ private MissingOutputTypeDescriptor(SerializableFunction<InputT, OutputT> fn) {
       this.fn = fn;
     }
 
-    public MapElements<InputT, OutputT> withOutputType(TypeDescriptor<OutputT> outputType) {
-      return new MapElements<>(fn, outputType);
+    public MapElements<InputT, OutputT> withOutputType(final TypeDescriptor<OutputT> outputType) {
+      return new MapElements<>(
+          SimpleFunction.fromSerializableFunctionWithOutputType(fn, outputType), fn.getClass());
     }
+
   }
 
   ///////////////////////////////////////////////////////////////////
 
-  private final SerializableFunction<InputT, OutputT> fn;
-  private final transient TypeDescriptor<OutputT> outputType;
+  private final SimpleFunction<InputT, OutputT> fn;
+  private final Class<?> fnClass; // for display data purposes
```

Comment:
Thank you! I did not identify the cause of the failure, but this was certainly the issue. Whatever I store does have to be serialized and non-transient, because of the unfortunate issue with `ParDo`. Your idea solved it. I'm going to drop the commit from this PR, please review #757 that is focused on just that.